src/evalfunc.c

```diff
@@ -6189,6 +6195,21 @@ f_line2byte(typval_T *argvars UNUSED, typval_T *rettv)
 #endif
 }
 
+/*
+ * "lshift()" function
+ */
+    static void
+f_lshift(typval_T *argvars, typval_T *rettv)
+{
+    if (argvars[0].v_type != VAR_NUMBER || argvars[1].v_type != VAR_NUMBER)
+    {
+	emsg(_(e_invarg));
+	return;
+    }
+    rettv->vval.v_number = tv_get_number(&argvars[0])
+				<< tv_get_number(&argvars[1]);
```

Comment:
This will result in UB (undefined behavior) if we do e.g. `lshift(1, 100)` as in C, it is UB to bit-shifts by more bits than the number of bits in the number. 

In Vim script, we don't want such undefined behavior.  I'd say `lshift(1, 100)` should return 0, and we must thus return 0 if argvars[1] is too high.  And we can add a test for bit shift with large number of bits.

We can verify that there is no runtime error using ubsan.

Also, what should happen when shifting by a negative number of bits? E.g. `lshift(1, -1)`  In C, it is undefined behavior but again, we don't want UB in Vim script.

Reply:
I will add a check for argvars[1] and return 0 if it is too large. If the shift value is negative, we can return the original value.